src/S3/MalformedResponseParser.php

```diff
@@ -0,0 +1,43 @@
+<?php
+namespace Aws\S3;
+
+use Aws\Api\Parser\AbstractParser;
+use Aws\CommandInterface;
+use Aws\Exception\AwsException;
+use Exception;
+use Psr\Http\Message\ResponseInterface;
+
+class MalformedResponseParser extends AbstractParser
+{
+    /** @var callable */
+    private $parser;
+    /** @var string */
+    private $exceptionClass;
+
+    public function __construct(
+        callable $parser,
+        $exceptionClass = AwsException::class
+    ) {
+        $this->parser = $parser;
+        $this->exceptionClass = $exceptionClass;
+    }
+
+    public function __invoke(
+        CommandInterface $command,
+        ResponseInterface $response
+    ) {
+        $fn = $this->parser;
+
+        try {
+            return $fn($command, $response);
+        } catch (Exception $e) {
```

Comment:
If this is meant to catch errors parsing XML, then I think catching `\Exception` is a little too broad. In fact, we should only be catching XML parsing errors to ensure that only that specific error is being retried. If we just catch all parsing errors, then we'd be retrying everything multiple times when we fail to parse. Instead of a catch-all like this, I think we should hand-pick what parsing errors are retryable (i.e., XML errors).